src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIService.java

```diff
@@ -0,0 +1,269 @@
+import de.tum.in.www1.artemis.domain.enumeration.ProjectType;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.exception.BambooException;
+import de.tum.in.www1.artemis.exception.LocalCIException;
+import de.tum.in.www1.artemis.repository.BuildLogStatisticsEntryRepository;
+import de.tum.in.www1.artemis.repository.FeedbackRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingSubmissionRepository;
+import de.tum.in.www1.artemis.service.BuildLogEntryService;
+import de.tum.in.www1.artemis.service.connectors.*;
+import de.tum.in.www1.artemis.service.dto.AbstractBuildResultNotificationDTO;
+import de.tum.in.www1.artemis.service.hestia.TestwiseCoverageService;
+
+@Service
+@Profile("localci")
+public class LocalCIService extends AbstractContinuousIntegrationService {
+
+    private final Logger log = LoggerFactory.getLogger(LocalCIService.class);
+
+    public LocalCIService(ProgrammingSubmissionRepository programmingSubmissionRepository, FeedbackRepository feedbackRepository, BuildLogEntryService buildLogService,
+            TestwiseCoverageService testwiseCoverageService, BuildLogStatisticsEntryRepository buildLogStatisticsEntryRepository) {
+        super(programmingSubmissionRepository, feedbackRepository, buildLogService, buildLogStatisticsEntryRepository, testwiseCoverageService);
+    }
+
+    @Override
+    public void createBuildPlanForExercise(ProgrammingExercise programmingExercise, String planKey, VcsRepositoryUrl sourceCodeRepositoryURL, VcsRepositoryUrl testRepositoryURL,
+            VcsRepositoryUrl solutionRepositoryURL) {
+        // TODO: Empty implementation to allow usage of 'localvc' with 'localci' in testing.
+    }
+
+    @Override
+    public void recreateBuildPlansForExercise(ProgrammingExercise exercise) {
+        deleteBuildPlan(exercise.getProjectKey(), exercise.getTemplateBuildPlanId());
+        deleteBuildPlan(exercise.getProjectKey(), exercise.getSolutionBuildPlanId());
+        createBuildPlanForExercise(exercise, BuildPlanType.TEMPLATE.getName(), exercise.getVcsTemplateRepositoryUrl(), exercise.getVcsTestRepositoryUrl(),
+                exercise.getVcsSolutionRepositoryUrl());
+        createBuildPlanForExercise(exercise, BuildPlanType.SOLUTION.getName(), exercise.getVcsSolutionRepositoryUrl(), exercise.getVcsTestRepositoryUrl(),
+                exercise.getVcsSolutionRepositoryUrl());
+    }
+
+    @Override
+    public void configureBuildPlan(ProgrammingExerciseParticipation participation, String branch) {
+        // TODO: Empty implementation to allow usage of 'localvc' with 'localci' in testing.
+        // TODO: Set publishBuildPlanUrl to false for all programming exercises created for local VC and local CI.
+    }
+
+    @Override
+    public void performEmptySetupCommit(ProgrammingExerciseParticipation participation) {
+        // Not needed for local CI. Implemented for Bamboo as a bug workaround.
```

Comment:
```suggestion
        // Not needed for local CI. Implemented for Bamboo as a bug workaround.
        throw new UnsupportedOperationException();
```
Maybe?

Reply:
This would throw an exception every time a programming exercise is started. The implementation in the `JenkinsService` is also just empty.